dvc_webdav/__init__.py

```diff
@@ -67,12 +67,19 @@ def _prepare_credentials(self, **config):
         headers = {}
         token = config.get("token")
         auth = None
+
+        custom_auth_header = config.get("custom_auth_header")
+
         if token:
             headers.update({"Authorization": f"Bearer {token}"})
         elif user:
             if not password and config.get("ask_password"):
                 password = ask_password(config["host"], user)
             auth = (user, password)
+        elif custom_auth_header:

```

Comment:
nitpicky suggestion to use walrus instead. WDYT? 
```suggestion
        elif custom_auth_header := config.get("custom_auth_header"):
```

Reply:
I like the suggestion and edited the code accordingly.